Integrations/python/deephaven/Types.py

```diff
@@ -0,0 +1,311 @@
+    :param instance: the object to which the wrapped function was bound when it was called
+    :param args: the argument list for `wrapped`
+    :param kwargs: the keyword argument dictionary for `wrapped`
+    :return: the decorated version of the method
+    """
+
+    _defineSymbols()
+    return wrapped(*args, **kwargs)
+
+try:
+    _defineSymbols()
+except Exception as e:
+    pass
+
+
+# Represents a Deephaven column data type.
+DataType = NewType('DataType', _qst_type_)
+
+# For more involved types, you can always use the string representation
+# of the Java class (Class.getName()) to get a python type for it.
+@_passThrough
+def typeFromJavaClassName(name : str):
+    """
+    Get the column data type for the corresponding Java type string reprensentation
+    The string provided should match the output in Java for Class.getName()
+    for a class visible to the main ClassLoader in the Deephaven engine in use.
+    """
+    jclass = _table_tools_.typeFromName(name)
+    return DataType(_qst_type_.find(jclass))
+
+
+#
+# Basic Deephaven column data types.
+# Column data types in python are represented as the jpy wrapper for the
+# corresponding Java class object for the column's Java type.
+#
+bool_ = DataType(_qst_type_.booleanType())
+byte = DataType(_qst_type_.byteType())
+short = DataType(_qst_type_.shortType())
+int16 = short  # make life simple for people who are used to pyarrow
+int_ = DataType(_qst_type_.intType())
+int32 = int_  # make life simple for people who are used to pyarrow
+long_ = DataType(_qst_type_.longType())
+int64 = long_   # make life simple for people who are used to pyarrow
+float_ = DataType(_qst_type_.floatType())
+single = float_   # make life simple for people who are used to NumPy
+float32 = float_  # make life simple for people who are used to pyarrow
+double = DataType(_qst_type_.doubleType())
+float64 = double  # make life simple for people who are used to pyarrow
+string = typeFromJavaClassName('java.lang.String')
+bigdecimal = typeFromJavaClassName('java.math.BigDecimal')
+stringset =  typeFromJavaClassName('io.deephaven.db.tables.libs.StringSet')
+datetime = DataType(_qst_type_.instantType())
+
+byte_array = typeFromJavaClassName('byte[]')
+short_array = typeFromJavaClassName('short[]')
+int16_array = short_array
+int_array = typeFromJavaClassName('int[]')
+int32_array = int_array
+long_array = typeFromJavaClassName('long[]')
+int64_array = long_array
+float_array = typeFromJavaClassName('float[]')
+single_array = float_array
+float32_array = float_array
+double_array = typeFromJavaClassName('double[]')
+float64_array = double_array
+string_array = typeFromJavaClassName('java.lang.String[]')
+
+@_passThrough
+def _jclassFromType(data_type : DataType):
+    if data_type is None:
+        return None
+    try:
+        jclass = data_type.clazz()
+    except Exception as e:
+        raise Exception("Could not get java class type from " + str(data_type)) from e
```

Comment:
Have you seen this happen? I think `qst.type.Type#clazz` should always return successfully.